File: projects/inflection/modules/dev/dev_py/ru/declension/init/parse/angle_brackets.py
from projects.inflection.modules.dev.dev_py import a
from projects.inflection.modules.dev.dev_py import mw
from projects.inflection.modules.dev.dev_py import tools as _

dev_prefix = 'User:Vitalik/'  # comment this on `prod` version


from ...init.parse import init_stem as init_stem
from ...init.parse import noun as noun_parse
from ...run.result import error as e


module = 'init.parse.angle_brackets'  # local


@a.starts(module)
def angle_brackets(func, i):
    angle_index = _.extract(i.rest_index, '%<([^>]+)%>')  # local
    if angle_index:
        if not i.pt:
            i.output_gender = i.gender
            i.output_animacy = i.animacy
        # end

        i.orig_index = i.index
        i.index = angle_index

        pt_backup = i.pt  # local
        noun_parse.extract_gender_animacy(i)
        i.pt = pt_backup

        if e.has_error(i):
            return _.ends(module, func)
        # end

        _.log_value(i.adj, 'i.adj')
        if i.adj:  # fixme: Для прилагательных надо по-особенному?
            init_stem.init_stem(i)
            if e.has_error(i):
                return _.ends(module, func)
            # end
        # end
    # end

    _.ends(module, func)
# end


# return export

File: projects/inflection/modules/dev/dev_lua/ru/declension/init/parse/angle_brackets.lua
local dev_prefix = ''
dev_prefix = 'User:Vitalik/'  -- comment this on `prod` version

local export = {}
local _ = require('Module:' .. dev_prefix .. 'inflection/tools')


local init_stem = require('Module:' .. dev_prefix .. 'inflection/ru/declension/init/parse/init_stem')  -- '..'
local noun_parse = require('Module:' .. dev_prefix .. 'inflection/ru/declension/init/parse/noun')  -- '..'
local e = require('Module:' .. dev_prefix .. 'inflection/ru/declension/run/result/error')  -- '..'


local module = 'init.parse.angle_brackets'


-- @starts
function export.angle_brackets(i)
	func = "angle_brackets"
	_.starts(module, func)

	local angle_index = _.extract(i.rest_index, '%<([^>]+)%>')
	if angle_index then
		if not i.pt then
			i.output_gender = i.gender
			i.output_animacy = i.animacy
		end

		i.orig_index = i.index
		i.index = angle_index

		local pt_backup = i.pt
		noun_parse.extract_gender_animacy(i)
		i.pt = pt_backup

		if e.has_error(i) then
			return _.ends(module, func)
		end

		_.log_value(i.adj, 'i.adj')
		if i.adj then  -- fixme: Для прилагательных надо по-особенному?
			init_stem.init_stem(i)
			if e.has_error(i) then
				return _.ends(module, func)
			end
		end
	end

	_.ends(module, func)
end


return export
